src/Glpi/Search/FilterableTrait.php

```diff
@@ -90,7 +90,14 @@ public function itemMatchFilter(CommonDBTM $item): bool
             'criteria' => $criteria,
         ]);
 
-        return $data['data']['totalcount'] > 0;
+        if ($data['data']['totalcount'] > 0) {
+            if (!isset($data['data']['rows'][0]['id'])) {
+                return false;
+            }
+            return true;
+        }
+
+        return false;
```

Comment:
1) A comment in the code to explain why this check is necessary would be appreciated.

2) I am not sure, but I guess that it is preferable to rely on `$id_field` instead of using an hardcoded `id` field name.

3) Is it possible to have both some invalid rows and some valid rows in the same result ? Maybe using `array_filter` to remove row that do not have a valid ID and then count them would be safer.

Reply:
[5fef5e6](https://github.com/glpi-project/glpi/pull/18289/commits/5fef5e601b82fd84cf60fab955428dd73ef27ef6)